terraform/aws/modules/cypher/main.tf

```diff
@@ -0,0 +1,111 @@
+terraform {
+  required_providers {
+    http = {
+      source  = "hashicorp/http"
+      version = "3.4.5"
+    }
+  }
+}
+
+data "http" "query_1" {
+  url    = local.db_url
+  method = "POST"
+
+  request_headers = {
+    Authorization = "Basic ${local.encoded_credentials}"
+    Content-Type  = "application/json"
+  }
+
+  # MERGE (n:NiamLastReleaseDate {date: "1997-01-01T23:59:59.000Z"}) return n;
+  request_body = jsonencode(
+    {
+      "statement" : "MERGE (n:NiamLastReleaseDate {date: $date}) RETURN n",
+      "parameters" : {
+        "date" : "1997-01-01T23:59:59.000Z"
+      }
+    }
+  )
+}
+
+
+data "http" "query_2" {
+  url    = local.db_url
+  method = "POST"
+
+  request_headers = {
+    Authorization = "Basic ${local.encoded_credentials}"
+    Content-Type  = "application/json"
+  }
+
+  # CALL apoc.create.uuids(5) YIELD uuid WITH collect(uuid) AS uuids
+  request_body = jsonencode(
+    {
+
+      "statement" = "CALL apoc.create.uuids(5) YIELD uuid WITH collect(uuid) AS uuids MERGE (client1:NiamClient {name: \"3edgesUIClient\"}) ON CREATE SET client1._id = uuids[0] MERGE (client2:NiamClient {name: \"3edgesConfigClient\"}) ON CREATE SET client2._id = uuids[1] MERGE (client3:NiamClient {name: \"3edgesDataproxyClient\"}) ON CREATE SET client3._id = uuids[2] MERGE (client4:NiamClient {name: \"3edgesDataloaderClient\"}) ON CREATE SET client4._id = uuids[3] MERGE (client5:NiamClient {name: \"3edgesDataproxyAuthzClient\"}) ON CREATE SET client5._id = uuids[4] WITH client1, client2, client3, client4, client5 MATCH (n:NiamClient) SET n.grant_types = [\"password\",\"implicit\",\"device_code\",\"social\",\"authorization_code\",\"client_credentials\",\"refresh_token\"], n.redirect_uris = [\"http://localhost:3005\", \"http://localhost:3005/code\",\"https://${var.hosted_zone}\", \"https://${var.hosted_zone}/code\", \"https://webloader.${var.hosted_zone}/code\"], n.post_logout_redirect_uris = [\"http://localhost:3005\",\"http://localhost:3005/login\", \"https://${var.hosted_zone}\", \"https://webloader.${var.hosted_zone}\", \"https://${var.hosted_zone}/login\", \"https://webloader.${var.hosted_zone}/login\"], n.response_types = [\"id_token\",\"code\",\"code id_token\"], n.token_endpoint_auth_method = \"client_secret_basic\", n.client_secret = \"U2FsdGVkX1+BhmwNe9+bzpTlx6/JHyghIyhxeDYPlcQ=\", n.client_id = n.name"
```

Comment:
I'm seeing a password here: "n.client_secret" 
You must use this value direct from your variables, right?

Reply:
I have made changes